File: Excel/700-799/774/774 Challenge.py
import pandas as pd

path = "700-799/774/774 Names Having Common Words.xlsx"
input = pd.read_excel(path, usecols="A", skiprows=1, nrows=20, names=["Names"])
test = pd.read_excel(path, usecols="C:D", skiprows=1, nrows=16, names=["Words", "Names"])\
    .sort_values(["Words", "Names"]).reset_index(drop=True)

common_words = pd.Series(' '.join(input['Names']).split()).value_counts()
common_words = common_words[common_words > 1].index.tolist()

result = pd.DataFrame([(word, name) for word in common_words for name in input['Names'] if word in str(name)],
                      columns=['Words', 'Names'])
result = result.sort_values(['Words', 'Names']).reset_index(drop=True)

print(result.equals(test)) # True

File: Excel/700-799/774/774 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/700-799/774/774 Names Having Common Words.xlsx"
input = read_excel(path, range = "A2:A22")
test  = read_excel(path, range = "C2:D18") %>% arrange(Words, Names)

common_words = input %>%
  mutate(id = row_number()) %>%
  separate_rows(Names, sep = " ") %>%
  count(Names, sort = TRUE) %>%
  filter(n > 1) %>%
  pull(Names)

result = expand.grid(common_words, input$Names, stringsAsFactors = F) %>%
  filter(str_detect(Var2, Var1)) %>%
  arrange(Var1, Var2)

all.equal(result, test, check.names = FALSE, check.attributes = FALSE)
#<> [1] TRUE